### PR TITLE
[None][chore] Update the Flux autodeploy example

### PR DESCRIPTION
**Pull request by ajrasane**
## Summary by CodeRabbit

* **New Features**
  * Added `--max_batch_size` CLI parameter to configure maximum batch size for the model.
  * Introduced a wrapper module for compiled models to provide consistent integration with existing pipelines.

* **Refactor**
  * Updated quantized model loading process.
  * Streamlined main execution flow for model compilation and initialization.

## Description

Updated the Flux example to be compatible with the latest version of auto deploy

## Test Coverage

TODO: Add integration test

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- The reviewers assigned automatically/manually are appropriate for the PR.

- [] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

`/bot [-h] ['run', 'kill', 'skip', 'reuse-pipeline'] ...`

Provide a user friendly way for developers to interact with a Jenkins server.

Run `/bot [-h|--help]` to print this help message.

See details below for each supported subcommand.

<details>

`run  [--reuse-test (optional)pipeline-id --disable-fail-fast --skip-test --stage-list "A10-PyTorch-1, xxx" --gpu-type "A30, H100_PCIe" --test-backend "pytorch, cpp" --add-multi-gpu-test --only-multi-gpu-test --disable-multi-gpu-test --post-merge --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx" --detailed-log --debug(experimental)]`

Launch build/test pipelines. All previously running jobs will be killed.

`--reuse-test (optional)pipeline-id ` *(OPTIONAL)* : Allow the new pipeline to reuse build artifacts and skip successful test stages from a specified pipeline or the last pipeline if no pipeline-id is indicated. If the Git commit ID has changed, this option will be always ignored. The DEFAULT behavior of the bot is to reuse build artifacts and successful test results from the last pipeline.

`--disable-reuse-test ` *(OPTIONAL)* : Explicitly prevent the pipeline from reusing build artifacts and skipping successful test stages from a previous pipeline. Ensure that all builds and tests are run regardless of previous successes.

`--disable-fail-fast ` *(OPTIONAL)* : Disable fail fast on build/tests/infra failures.

`--skip-test ` *(OPTIONAL)* : Skip all test stages, but still run build stages, package stages and sanity check stages. Note: Does **NOT** update GitHub check status.

`--stage-list "A10-PyTorch-1, xxx"` *(OPTIONAL)* : Only run the specified test stages. Examples: "A10-PyTorch-1, xxx". Note: Does **NOT** update GitHub check status.

`--gpu-type "A30, H100_PCIe"` *(OPTIONAL)* : Only run the test stages on the specified GPU types. Examples: "A30, H100_PCIe". Note: Does **NOT** update GitHub check status.

`--test-backend "pytorch, cpp"` *(OPTIONAL)* : Skip test stages which don't match the specified backends. Only support [pytorch, cpp, tensorrt, triton]. Examples: "pytorch, cpp" (does not run test stages with tensorrt or triton backend). Note: Does **NOT** update GitHub pipeline status.

`--only-multi-gpu-test ` *(OPTIONAL)* : Only run the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--disable-multi-gpu-test ` *(OPTIONAL)* : Disable the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--add-multi-gpu-test ` *(OPTIONAL)* : Force run the multi-GPU tests in addition to running L0 pre-merge pipeline.

`--post-merge ` *(OPTIONAL)* : Run the L0 post-merge pipeline instead of the ordinary L0 pre-merge pipeline.

`--extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx"` *(OPTIONAL)* : Run the ordinary L0 pre-merge pipeline and specified test stages. Examples: --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx".

`--detailed-log ` *(OPTIONAL)* : Enable flushing out all logs to the Jenkins console. This will significantly increase the log volume and may slow down the job.

`--debug ` *(OPTIONAL)* : **Experimental feature**. Enable access to the CI container for debugging purpose. Note: Specify exactly one stage in the `stage-list` parameter to access the appropriate container environment. Note: Does **NOT** update GitHub check status.

For guidance on mapping tests to stage names, see `docs/source/reference/ci-overview.md`
and the `scripts/test_to_stage_mapping.py` helper.

### kill

`kill  `

Kill all running builds associated with pull request.

### skip

`skip --comment COMMENT `

Skip testing for latest commit on pull request. `--comment "Reason for skipping build/test"` is required. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

### reuse-pipeline

`reuse-pipeline `

Reuse a previous pipeline to validate current commit. This action will also kill all currently running builds associated with the pull request. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

</details>

### CODE REVIEW

**Review by juney-nvidia — APPROVED** [review]

File: examples/auto_deploy/build_and_run_flux.py [hunk 1]
```diff
@@ -115,48 +223,122 @@ def main():
...
+    optimizer = InferenceOptimizer(factory=None, config=optimizer_config)
+    gm = optimizer(cm=None, mod=gm)
+    ad_logger.info("Inference optimizer fusions applied successfully")
 
-    compiler_cls = CompileBackendRegistry.get("torch-opt")
-    gm = compiler_cls(gm, args=(), kwargs=flux_kwargs).compile()
+    # Validate backend availability
+    if not CompileBackendRegistry.has(args.backend):
+        available = CompileBackendRegistry.list()
+        raise ValueError(f"Backend '{args.backend}' not found. Available backends: {available}")
```
Comment (lucaslie) [inline examples/auto_deploy/build_and_run_flux.py]: You should be able to just add the compile transform to your optimizer_config - then you don't have to call it manually here. Here is how it's configured: https://github.com/NVIDIA/TensorRT-LLM/blob/b26e1617f2fc648e22fa185ff14c33ae8587a385/tensorrt_llm/_torch/auto_deploy/config/default.yaml#L168-L172

File: requirements.txt [hunk 2]
```diff
@@ -74,3 +74,4 @@ nvidia-cutlass-dsl==4.2.1; python_version >= "3.10"
 plotly
 numexpr<2.14.0 # WAR for attempted use of nonexistent numpy.typing
 partial_json_parser
+tqdm
```
Comment (lucaslie) [inline requirements.txt]: we should probably not add requirements here if it's only used in examples. Instead you can provide instructions in the examples to pip install this

File: tests/integration/defs/examples/test_flux.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    PROMPT = "a photo of an astronaut riding a horse on mars"
+    MIN_CLIP_SIMILARITY = 0.25
+    NUM_INFERENCE_STEPS = 20
+    MAX_BATCH_SIZE = 1
+    BACKEND = "torch-opt"
+
+    # Checkpoint paths for different quantization formats
+    # These can be set via environment variables or test parameters
+    FP8_CHECKPOINT = os.environ.get("FLUX_FP8_CHECKPOINT")
+    FP4_CHECKPOINT = os.environ.get("FLUX_FP4_CHECKPOINT")
```
Comment (VALLIS-NERIA) [inline tests/integration/defs/examples/test_flux.py]: For example, if you uploaded the checkpoint to `llm-models/flux/fp4`, you can use `FP4_CHECKPOINT = f'llm_models_root()/flux/fp4'

File: examples/auto_deploy/build_and_run_flux.py (hunk 1 above)
Comment (ajrasane) [reply examples/auto_deploy/build_and_run_flux.py]: Done

File: tensorrt_llm/_torch/auto_deploy/utils/_graph.py [hunk 4]
```diff
@@ -65,6 +65,10 @@ def load_buffers_and_params(
                 v_new = nn.Parameter(v_new, requires_grad=False)
         else:
             v_new = state_dict[k]
+            # Check if the original attribute is a Parameter and wrap accordingly
+            original_attr = getattr(submod, name, None)
+            if isinstance(original_attr, nn.Parameter) and not isinstance(v_new, nn.Parameter):
+                v_new = nn.Parameter(v_new, requires_grad=False)
```
[the remark this answers is not in the document]
Comment (ajrasane) [reply tensorrt_llm/_torch/auto_deploy/utils/_graph.py]: This was required while loading the real-quantized checkpoint. But we wont need this while loading the fake quantized checkpoint. Removed.

File: examples/auto_deploy/flux_transforms.yaml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    stage: "post_load_fusion"
+    backend: "torch"
+  fuse_nvfp4_linear:
+    stage: "post_load_fusion"
+    backend: "trtllm"
+  fuse_fp8_gemms:
+    stage: "post_load_fusion"
+  fuse_fp4_gemms:
+    stage: "post_load_fusion"
+
```
[the remark this answers is not in the document]
Comment (Fridah-nv) [reply examples/auto_deploy/flux_transforms.yaml]: But I think we can hold back these optimization transforms for this PR since there's accuracy issue with the passes on FLUX.

**Review by Fridah-nv — APPROVED** [review]
